woocommerce/woocommerce-ios
# Subscription Products: Add sign-up fee to the price row and screen

Closes: #11135 


## What

This PR adds the sign-up fee row for subscription products on the price settings screen. The price row in the product form has also been updated to display the sign-up fee if it's available.

## How
- Displayed the sign-up fee information in the price row in the product form if it's available.
- Added a new row type and showed it on the price settings screen for subscription products.
- Updated logic to include sign-up fee when checking for changes on the price settings screen.
- Handled sign-up fee updates on the price settings screen.
- Updated the callback closure for the price settings screen to include sign-up fee.
- Updated unit tests.

## Testing instructions

- Enable the feature flag `subscriptionProducts` to be able to edit subscription products.
- Create a simple subscription product for your test site on wp-admin if you haven't already.
- Build the app, log in to your test store and navigate to the Products tab.
- Select a simple subscription product.
- If the product has a sign-up fee, the fee should be displayed on the price row of the product form.
- Tap the price row, notice a few changes:
  - A new row for subscription sign-up fee is displayed below the subscription period row. 
  - Change the sign-up fee, tap on Done on the price settings screen, and tap Save on the product form. The new fee should be updated correctly.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/e31d68d5-66f3-43b6-b7a9-9d6bde4957bf



---
